public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -279,6 +283,15 @@ export function PrometheusDatasource(instanceSettings, $q, backendSrv, templateS
     return metricName + '{' + labelPart + '}';
   };
 
+  this.alignTime = function(time, range) {
```

Comment:
No has nothing to do with refresh period, but time window size, The alignment should not change the time window by a large margin